testsuite/forge/src/backend/k8s/mod.rs

```diff
@@ -21,6 +25,56 @@ impl K8sFactory {
             treasury_compliance_key,
         }
     }
+
+    // Instead of having the root_key and treasury_compliance_key explicitly, we should
+    // fetch these from the cluster internally via Vault, assuming the cluster environment
+    // has provided access
+    pub fn new_internal() -> std::result::Result<K8sFactory, Error> {
+        let vault_addr = env::var("VAULT_ADDR")
+            .map_err(|_| format_err!("Expected environment variable VAULT_ADDR"))?;
+        let vault_cacert = env::var("VAULT_CACERT")
+            .map_err(|_| format_err!("Expected environment variable VAULT_CACERT"))?;
+        let vault_token = env::var("VAULT_TOKEN")
+            .map_err(|_| format_err!("Expected environment variable VAULT_TOKEN"))?;
```

Comment:
Maybe we want to include some documentation about what these env vars are and how to get them (e.g. in the runbook)

Reply:
These are provided by the environment in the pod before `forge` runs, but yea I can document this. https://github.com/diem/partners/blob/master/testnet/forge/files/init_forge.sh